module/userProjects/helper.js

```diff
@@ -2362,21 +2362,70 @@ module.exports = class UserProjectsHelper {
 	}
 
 	/**
-	 * List of projects.
+	 * List of user projects.
 	 * @method
 	 * @name list
+	 * @param {String} userId -UserID
+	 * @param {Number} pageNo -Page number
+	 * @param {Number} pageSize -Page size
+	 * @param {String} searchText -Search text
+	 * @param {String} language -LanguageCode
 	 * @returns {Array} List of projects.
 	 */
 
-	static list(bodyData, pageNo, pageSize, searchText, filter) {
+	static list(userId, pageNo, pageSize, searchText, language = '') {
 		return new Promise(async (resolve, reject) => {
 			try {
-				let projects = await projectQueries.projectDocument(
-					bodyData.query,
-					bodyData.projection,
-					bodyData.skipFields
+				let aggregateData = []
+				// Match Query
+				let matchQuery = {
+					$match: {
+						userId: userId,
+						status: { $ne: CONSTANTS.common.COMPLETED_STATUS },
+					},
+				}
+				aggregateData.push(matchQuery)
+
+				// Projection aggregate for multilingual
+				let titleField = language ? `$translations.${language}.title` : '$title'
+				let descriptionField = language ? `$translations.${language}.description` : '$description'
+				aggregateData.push({
+					$project: {
+						_id: 1,
+						title: { $ifNull: [titleField, '$title'] },
+						description: { $ifNull: [descriptionField, '$description'] },
+						projectTemplateId: 1,
+						projectTemplateExternalId: 1,
+						status: 1,
+						tasks: 1,
```

Comment:
@praveenKDass what about translation  of task data? it will also happen?

Reply:
Yes it will happen on lino 2426 which we need to get task translate on details as well @VISHNUDAS-tunerlabs